### PR TITLE
Feature/new look grants

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
- star/unstar a grant using vue
- list view of all the grants,
- Fix grant nav sort issue 

https://www.loom.com/share/8eec68f5f2d243c58968f9bd2db25a96

##### Refers/Fixes
https://github.com/gitcoinco/web/issues/7163
https://github.com/gitcoinco/web/issues/7166
https://github.com/gitcoinco/web/issues/7190

### CODE REVIEW

**Comment by zoek1** [comment]
@thelostone-mc are you have some time to check this please? 🙂

File: app/grants/templates/grants/shared/landing_grants.html [hunk 1]
```diff
@@ -2,38 +2,33 @@
...
     <div class="container-fluid grants-container">
       <div class="container">
-        {% if grants %}
-          <div class="infinite-container row">
-            {% for grant in grants %}
-              <div class="col-12 col-md-6 col-xl-4 mb-4 infinite-item">
-                {% include 'grants/card/index.html' %}
-              </div>
-            {% endfor %}
+        <div class="infinite-container row" v-if="grants">
```
Comment (octavioamu) [inline app/grants/templates/grants/shared/landing_grants.html:5]: I think scroll method is not being call here so probably pagination will not work you need to use the directive `scroll.passive` (check notifications.html)

File: app/grants/templates/grants/index.html [hunk 2]
```diff
@@ -56,35 +56,37 @@
               </div>
             </div>
             {% endif %}
-            <div class="row">
-              <div class="col-12 col-lg-3 col-xl-2" id="sidebar_container">
-                {% include 'grants/shared/sidebar_search.html' %}
-              </div>
+            <div id="grants-showcase">
```
Comment (octavioamu) [inline app/grants/templates/grants/index.html:59]: missing v-cloak to hide the vue components until data is loaded (avoid reflow)

File: app/grants/templates/grants/shared/landing_grants.html [hunk 3]
```diff
@@ -2,38 +2,33 @@
...
           </div>
-          {% if grants.has_next %}
-            <p class="mb-0 text-center py-4">
-              <a class="infinite-more-link text-center text-black-70 cursor-pointer"
-                href="?network={{network}}&type={{type}}&keyword={{keyword}}&sort_option={{sort}}&page={{ grants.next_page_number }}"
-              >
-                <i class="fas fa-spinner fa-spin mr-2"></i>
+        </div>
+
+        <!--p class="mb-0 text-center py-4" v-if="has_next">
```
Comment (octavioamu) [inline app/grants/templates/grants/shared/landing_grants.html:11]: dead code

**Review by octavioamu — CHANGES_REQUESTED** [review]
Well done Zoek, I left just some few comments

**Comment by thelostone-mc** [comment]
@zoek1 could you point this to the integration branch https://github.com/gitcoinco/web/tree/round-7-integration

**Comment by thelostone-mc** [comment]
Also could you show 

- how the page looks in list view when there is no active CLR
- could you rebase with integration commits

File: app/grants/templates/grants/card/front.html [hunk 4]
```diff
@@ -64,6 +65,17 @@ <h2 class="grant-item__title font-subheader"><a href="{% url 'grants:details' gr
...
+        {% if profile.user.is_authenticated %}
+        {% is_favorite grant profile.user as is_grant_favorite %}
+        <button class="text-decoration-none btn btn-link star-action m-0 p-0 mr-4 mt-3 gc-text-blue {% if not is_grant_favorite %}text-muted{% endif %}" data-grant="{{ grant.id }}">
+          {% if not is_grant_favorite %}
+            <i class="far fa-star" ></i><span class="ml-1 font-body">Follow</span>
+          {% else %}
+            <i class="fa fa-star" ></i><span class="ml-1 font-body">Following</span>
+          {% endif %}
+        </button>
+        {% endif %}
```
Comment (thelostone-mc) [inline app/grants/templates/grants/card/front.html:78]: ^ indent it right
```
{% if profile.user.is_authenticated %}
    ....
{% endif %}
```

File: app/grants/templates/grants/components/card.html [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+{% endcomment %}
+{% load i18n static humanize grants_extra %}
+<grant-card inline-template :grant="grant" :cred="credentials" :view="view">
+  <div class="grant-item grant-item--list flex-md-row flex-column" v-if="view == 'list'">
+      <div class="grant-item__img col-md-1 d-md-block d-none">
```
Comment (thelostone-mc) [inline app/grants/templates/grants/components/card.html:20]: ^ 1 tab= 2 spaces indentation please

File: app/grants/templates/grants/components/card.html [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+          </template>
+        </div>
+    </div>
+    <div class="grant-item__footer">
+      <template v-if="1">
+          <div class="clr mb-3">
+            <div class="match__round pb-1 pt-2 mb-2">
+              <h2 class="title col-8 font-caption g-bold">
+                  CLR MATCH ROUND 6
+              </h2>
```
Comment (thelostone-mc) [inline app/grants/templates/grants/components/card.html:170]: ^ why is this hardcoded ?
This is something which should be available from the backend API.

File: app/grants/templates/grants/components/card.html [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+                  </h2>
+                  <p class="amount mb-2 font-caption font-weight-semibold">MONTHLY RECURRING</p>
+                </template>
+            </div>
+          </div>
+
+
+          <div class="col-12 pb-3" v-if="grant.is_clr_eligible">
+            {# with short=True #}
+            {% include 'grants/card/clr_estimate.html'  %}
```
Comment (thelostone-mc) [inline app/grants/templates/grants/components/card.html:205]: shouldn't this be {% include 'grants/card/clr_estimate.html' with short=True %}

File: app/grants/templates/grants/components/card.html [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+
+            <button class="btn btn-gc-grey cart-btn large font-body font-weight-semibold mt-1 mr-1" v-else>
+              <i class="fa fa-stop" aria-hidden="true"></i> Cannot Fund your Own Grant
+            </button>
+          </div>
+        </template>
+    </div>
+  </a>
+    </div>
+</grant-card>
```
Comment (thelostone-mc) [inline app/grants/templates/grants/components/card.html:229]: ^ could you redo the text and the conditions based on what's in `front.html` 
this code seems a lil outdated

File: app/grants/templates/grants/card/front.html (hunk 4 above)
Comment (thelostone-mc) [reply app/grants/templates/grants/card/front.html:78]: Why are there changes in this file when this you've removed it's usage in 
https://github.com/gitcoinco/web/pull/7332/files#diff-a9ea52761a5188515132b584f9014d16L9 ?

**Review by thelostone-mc — CHANGES_REQUESTED** [review]
@zoek1 looking into this code ! A lot of things would break.
Looks like you started with an outdated branch (over a month old) and never rebased after that

I'd recommend taking the latest code and porting your css + model changes
and redoing the views + HTML changes with the latest code and ensuring you don't overwrite anything

**Pull request closed without merging**